core/src/main/java/tech/pegasys/web3signer/core/Runner.java

```diff
@@ -179,51 +168,82 @@ private void registerHttpHostAllowListHandler(final OpenAPI3RouterFactory openAp
+    addEth2PublicKeysListHandler(openAPI3RouterFactory, blsIdentifiers, errorHandler);
 
     // sign handler
-    final Counter missingSignerCounter =
-        metricsSystem.createCounter(
-            Web3SignerMetricCategory.SIGNING,
-            "missing_identifier_count",
-            "Number of signing operations requested, for keys which are not available");
+    addEth1SignHandler(openAPI3RouterFactory, secpSigner, metricsSystem, errorHandler);
+    addEth2SignHandler(openAPI3RouterFactory, blsSigner, metricsSystem, errorHandler);
+
+    // upcheck handler
+    addUpcheckHandler(openAPI3RouterFactory, errorHandler);
+
+    return openAPI3RouterFactory;
+  }
 
+  private void addEth2SignHandler(
+      final OpenAPI3RouterFactory openAPI3RouterFactory,
+      final SignerForIdentifier<BlsArtifactSignature> blsSigner,
+      final MetricsSystem metricsSystem,
+      final LogErrorHandler errorHandler) {
     openAPI3RouterFactory.addHandlerByOperationId(
-        SIGN_FOR_IDENTIFIER_OPERATION_ID,
+        ETH2_SIGN.name(),
         new BlockingHandlerDecorator(
-            new SignForIdentifierHandler(blsSigner, metricsSystem, "bls"), true));
+            new SignForIdentifierHandler(
+                blsSigner, new OpenApiOperationsMetrics(metricsSystem, BLS)),
+            false));
+    openAPI3RouterFactory.addFailureHandlerByOperationId(ETH2_SIGN.name(), errorHandler);
+  }
+
+  private void addEth1SignHandler(
+      final OpenAPI3RouterFactory openAPI3RouterFactory,
+      final SignerForIdentifier<SecpArtifactSignature> secpSigner,
+      final MetricsSystem metricsSystem,
+      final LogErrorHandler errorHandler) {
     openAPI3RouterFactory.addHandlerByOperationId(
-        SIGN_FOR_IDENTIFIER_OPERATION_ID,
+        ETH1_SIGN.name(),
         new BlockingHandlerDecorator(
-            new SignForIdentifierHandler(secpSigner, metricsSystem, "secp"), true));
+            new SignForIdentifierHandler(
+                secpSigner, new OpenApiOperationsMetrics(metricsSystem, SECP256K1)),
+            false));
+    openAPI3RouterFactory.addFailureHandlerByOperationId(ETH1_SIGN.name(), errorHandler);
+  }
+
+  private void addEth1PublicKeysListHandler(
+      final OpenAPI3RouterFactory openAPI3RouterFactory,
+      final Set<String> secpIdentifiers,
+      final LogErrorHandler errorHandler) {
     openAPI3RouterFactory.addHandlerByOperationId(
-        SIGN_FOR_IDENTIFIER_OPERATION_ID,
-        rc -> {
-          missingSignerCounter.inc();
-          rc.next();
-        });
-    openAPI3RouterFactory.addFailureHandlerByOperationId(
-        SIGN_FOR_IDENTIFIER_OPERATION_ID, errorHandler);
+        ETH1_LIST.name(),
+        new BlockingHandlerDecorator(new PublicKeysListHandler(secpIdentifiers), false));
+    openAPI3RouterFactory.addFailureHandlerByOperationId(ETH1_LIST.name(), errorHandler);
+  }
 
-    return openAPI3RouterFactory;
+  private void addEth2PublicKeysListHandler(
```

Comment:
similar to the above comment, looks like some dupe here, these publicKeysListHandlers are exactly the same except for some  naming

Reply:
refactored out to common logic.